File: Practice/18/Python/18. Евротур/18. Евротур/_18._Евротур.py
chars = [0] * 26
tochtoznaet = ("hallo", "klempner", "das", "ist", "fantastisch", "fluggegecheimen")
stopslovo = str(input("Стоп слово: "))
a = ord('a')
for word in tochtoznaet:
    for char in word:
        chars[ord(char) - a] += 1
probability = 1
rez = sum(chars)

for char in stopslovo:
    probability *= chars[ord(char) - a] / rez
    if (probability == 0): 
        break
print(probability)

File: Practice/18/C++/18. Евротур/18. Евротур/18. Евротур.cpp
#include <iostream>
#include <string>
using namespace std;
int main()
{
    setlocale(LC_ALL, "ru");
    const string ws[] = { "hallo", "klempner", "das", "ist", "fantastisch", "fluggegecheimen" };
    int chars[26] = {};
    const int i = 'a';
    double probability = 1; double sum = 0;
    cout << "Введите стоп слово:" << endl;
    string stopslovo; 
    getline(cin, stopslovo);
    for (const string& w : ws)
    {
        for (const char& chr : w)
        {
            ++chars[chr - i];
        }
    }
    for (const int& i : chars)
    {
        sum = sum + i;
    }
    for (const char& chr : stopslovo)
    {
        probability = probability * chars[chr - i] / sum;
        if (probability < LDBL_EPSILON)
            break;
    }
    cout << probability << endl;
}
